src/test/cypress/integration/exam/ExamDateVerification.spec.ts

```diff
@@ -0,0 +1,159 @@
+                .build();
+            courseManagementRequests.createExam(examContent).then((response) => {
+                exam = response.body;
+            });
+            cy.login(artemis.users.getStudentOne(), `/courses`);
+            cy.contains(examTitle).should('not.exist');
+            cy.visit(`/courses/${course.id}`);
+            cy.url().should('contain', `${course.id}`);
+            cy.contains(examTitle).should('not.exist');
+        });
+
+        it('Shows after visible date', () => {
+            const examContent = new CypressExamBuilder(course)
+                .title(examTitle)
+                .visibleDate(dayjs().subtract(5, 'days'))
+                .startDate(dayjs().add(2, 'days'))
+                .endDate(dayjs().add(3, 'days'))
+                .build();
+            courseManagementRequests.createExam(examContent).then((response) => {
+                exam = response.body;
+                courseManagementRequests.registerStudentForExam(course, exam, artemis.users.getStudentOne());
+                cy.login(artemis.users.getStudentOne(), `/courses/${course.id}`);
+                cy.url().should('contain', `${course.id}`);
+                cy.contains('Exams').click();
+                cy.url().should('contain', '/exams');
+                cy.contains(examTitle).should('exist').click();
+                cy.url().should('contain', `/exams/${exam.id}`);
+            });
+        });
+
+        it('Student can start after start Date', () => {
+            let exerciseGroup: any;
+            let textExercise: any;
+            const student = artemis.users.getStudentOne();
+            const examContent = new CypressExamBuilder(course)
+                .title(examTitle)
+                .visibleDate(dayjs().subtract(3, 'days'))
+                .startDate(dayjs().subtract(2, 'days'))
+                .endDate(dayjs().add(3, 'days'))
+                .build();
+            courseManagementRequests.createExam(examContent).then((examResponse) => {
+                exam = examResponse.body;
+                courseManagementRequests.registerStudentForExam(course, exam, student);
+                courseManagementRequests.addExerciseGroupForExam(course, exam, 'group 1', true).then((groupResponse) => {
+                    exerciseGroup = groupResponse.body;
+                    courseManagementRequests.addTextExerciseToExam(exerciseGroup, 'Text exercise 1').then((exerciseResponse) => {
+                        textExercise = exerciseResponse.body;
+                        courseManagementRequests.generateMissingIndividualExams(course, exam);
+                        courseManagementRequests.prepareExerciseStartForExam(course, exam);
+                        cy.login(student, `/courses/${course.id}/exams`);
+                        cy.contains(exam.title).click();
+                        cy.url().should('contain', `/exams/${exam.id}`);
+                        cy.contains('Welcome to ' + exam.title).should('exist');
```

Comment:
I think we should check the visibility here. It is a more strict assertion since everything that is visible also has to exist in the DOM. When starting an exam this should be visible without having to scroll there
```suggestion
                        cy.contains('Welcome to ' + exam.title).should('be.visible');
```